nannyml/base.py

```diff
@@ -124,6 +125,10 @@ def _get_property_for_key(self, key: Key, property_name: str) -> Optional[pd.Ser
 
 
 class Abstract1DResult(AbstractResult, ABC):
+    def __init__(self, results_data: pd.DataFrame, metrics: Sequence[Metric] = (), *args, **kwargs):
```

Comment:
Any particular reason to use Sequence over List here? Just making it more generic?

Reply:
Yes, this was necessary to satisfy the static type checker. Sequence is covariant, while List is not ([PEP484](https://peps.python.org/pep-0484/#covariance-and-contravariance) gives a description of the difference).

We're passing in instances that match the `Metric` `Protocol`, but not actual instances of `Metric`. `List[Metric]` doesn't type check as a result of that, but `Sequence[Metric]` does. Additionally, it has the benefit of being less restrictive (we can pass in tuples for example).